server/src/main/java/io/crate/expression/operator/all/AllEqOperator.java

```diff
@@ -46,11 +56,61 @@ boolean matches(Object probe, Object candidate) {
 
     @Override
     protected Query refMatchesAllArrayLiteral(Function all, Reference probe, Literal<?> literal, LuceneQueryBuilder.Context context) {
-        return null;
+        // col = ALL ([1,2,3]) --> col=1 and col=2 and col=3
+        var values = StreamSupport
+            .stream(((Iterable<?>) literal.value()).spliterator(), false)
+            .toList();
+        LinkedHashSet<?> uniqueValues = new LinkedHashSet<>(values);
```

Comment:
Can skip the first `.toList` and build a `HashSet` directly.
`LinkedHashSet` is not necessary given that the order doesn't matter.

Reply:
Thank you, updated.